### PR TITLE
Feature: support custom cell ellipsis tooltip 

### PR DESCRIPTION
**Pull request by lijinke666**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/react-component/table/pull/474

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Custom cell ellipsis tooltip       |
| 🇨🇳 Chinese |   自定义单元格省略后的提示      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered .github/PULL_REQUEST_TEMPLATE.md](https://github.com/lijinke666/ant-design/blob/feature/.github/PULL_REQUEST_TEMPLATE.md)
[View rendered .github/PULL_REQUEST_TEMPLATE/pr_cn.md](https://github.com/lijinke666/ant-design/blob/feature/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)
[View rendered components/table/demo/ellipsis-custom-tooltip.md](https://github.com/lijinke666/ant-design/blob/feature/components/table/demo/ellipsis-custom-tooltip.md)
[View rendered components/table/index.en-US.md](https://github.com/lijinke666/ant-design/blob/feature/components/table/index.en-US.md)
[View rendered components/table/index.zh-CN.md](https://github.com/lijinke666/ant-design/blob/feature/components/table/index.zh-CN.md)

### CODE REVIEW

**Comment by lijinke666** [comment]
本地测试跑不过, 没有动到这块呀？

image

**Comment by hengkx** [comment]
重新安装 删除lock也不行吗？ 这个 好像是我改的 ，昨天提pr是全部可以通过的

File: components/table/index.zh-CN.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ const columns = [
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
 | align | 设置列的对齐方式 | `left` \| `right` \| `center` | `left` |  |
-| ellipsis | 超过宽度将自动省略，暂不支持和排序筛选一起使用。<br />设置为 `true` 时，表格布局将变成 `tableLayout="fixed"`。 | boolean | false |  |
+| ellipsis | 超过宽度将自动省略，暂不支持和排序筛选一起使用。<br />设置为 `true` 或 `{ showTitle?: boolean }` 时，表格布局将变成 `tableLayout="fixed"`。 | boolean \| { showTitle?: boolean } | false |  |
```
Comment (hengkx) [inline components/table/index.zh-CN.md:122]: 加个版本号

Comment (lijinke666) [reply components/table/index.zh-CN.md:122]: 上哪辆车? 4.3.0?

**Comment by lijinke666** [comment]
> 重新安装 删除lock也不行吗？ 这个 好像是我改的 ，昨天提pr是全部可以通过的

我再试试

**Comment by lijinke666** [comment]
删除lock重新安装 ok 了

File: components/table/index.zh-CN.md (hunk 1 above)
Comment (hengkx) [reply components/table/index.zh-CN.md:122]: 4.3.0

**Review by afc163 — APPROVED** [review]

File: components/table/demo/ellipsis-custom-tooltip.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  {
+    title: 'Long Column Long Column',
+    dataIndex: 'address',
+    key: 'address 3',
+    ellipsis: {
+      showTitle: false,
+    },
+    render(address) {
+      return (
```
Comment (afc163) [inline components/table/demo/ellipsis-custom-tooltip.md:71]: 直接用 =>

File: components/table/index.zh-CN.md (hunk 1 above)
Comment (lijinke666) [reply components/table/index.zh-CN.md:122]: 版本号有个问题: ellipsis: boolean 是之前版本就有, { showTitle?: boolean } 是新加的, 这里直接写  4.3.0 是不是会有误解

File: components/table/demo/ellipsis-custom-tooltip.md (hunk 2 above)
Comment (lijinke666) [reply components/table/demo/ellipsis-custom-tooltip.md:71]: 好

File: components/table/index.zh-CN.md (hunk 1 above)
Comment (hengkx) [reply components/table/index.zh-CN.md:122]: image

**Comment by lijinke666** [comment]
@hengkx 可以了

**Comment by TnWah** [comment]
这样设置showTitle为false，自定义render Tooltip会出现一个问题就是可能某一行的数据并没有省略，内容可以显示的下的情况也会出现Tooltip，有没有优化方案呢？

**Pull request merged**